lib/elixir/lib/exception.ex

```diff
@@ -787,7 +787,14 @@ defmodule File.Error do
   defexception [:reason, :path, action: ""]
 
   def message(exception) do
-    formatted = IO.iodata_to_binary(:file.format_error(exception.reason))
+    formatted =
+      case {exception.action, exception.reason} do
```

Comment:
great suggestion.
I didn't create the exception variable though, as I see no use for it.